js/src/data/resolvers.js

```diff
@@ -402,3 +403,25 @@ getMappingRules.shouldInvalidate = ( action ) => {
 		action.type === TYPES.DELETE_MAPPING_RULE
 	);
 };
+
+/**
+ * Resolver for getting the Store categories.
+ */
+export function* getStoreCategories() {
+	try {
+		const response = yield apiFetch( {
+			path: `${ API_NAMESPACE }/mc/mapping/categories`,
+		} );
+
+		yield receiveStoreCategories( response );
+	} catch ( error ) {
+		yield handleFetchError(
+			error,
+			__(
+				'There was an error getting the store categories.',
+				'google-listings-and-ads'
+			)
+		);
+		throw error;
```

Comment:
Since the selectors of wp-data won't catch errors internally nor re-throw out them to somewhere for the library consumer side to catch them, this `throw` statement could be omitted.